File: 630-course-schedule-iii/630-course-schedule-iii.java
class Solution {
    public int scheduleCourse(int[][] courses) {
        var maxHeap = new PriorityQueue<Integer>(Comparator.reverseOrder());
        Arrays.sort(courses, Comparator.comparingInt(course -> course[1]));
        for(int i = 0, total = 0; i < courses.length; ++i) {
            maxHeap.add(courses[i][0]);
            if((total += courses[i][0]) > courses[i][1])
                total -= maxHeap.poll();
        }
        return maxHeap.size();
    }
}

File: 630-course-schedule-iii/630-course-schedule-iii.py
class Solution:
    def scheduleCourse(self, courses: List[List[int]]) -> int:
        heap, total = [], 0
        courses.sort(key = lambda x : x[1])
        for i in range (len(courses)):
            heappush(heap, -courses[i][0])
            total += courses[i][0]
            if total > courses[i][1]:
                total -= -heap[0]
                heappop(heap)
        return len(heap)